File: FileMan/src/main/java/com/example/FileMan/model/Project.java
package com.example.FileMan.model;


import jakarta.persistence.*;
import jakarta.validation.constraints.NotEmpty;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;

import static jakarta.persistence.GenerationType.AUTO;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@NamedStoredProcedureQueries({
        @NamedStoredProcedureQuery(name="getProjects", procedureName = "getProjects"),
        @NamedStoredProcedureQuery(name="createProject", procedureName = "createProject",
        parameters = {
                @StoredProcedureParameter(mode = ParameterMode.IN, name="name", type= String.class),
                @StoredProcedureParameter(mode = ParameterMode.IN, name="description", type= String.class),
                @StoredProcedureParameter(mode = ParameterMode.IN, name="createdBy", type= String.class),
                @StoredProcedureParameter(mode = ParameterMode.IN, name="accessKey", type= String.class),
        })
})
public class Project {
    @Id
    @GeneratedValue(strategy = AUTO)
    private Long pid;
    @Column(unique = true)
    @NotEmpty(message = "Must not be empty or null")
    private String name;
    private String description;
    private String createdBy;
    private Date createdDate;
    private String accessKey;

}

File: FileManFront/src/app/Demo/Module/project.ts
export interface Project{
    pid: number;
    name: string;
    description: string;
    createdBy: string;
    createdDate: Date;
    accessKey: String;
}


export interface ProjectExport{
    name: string;
    description: string;
    createdBy: String;
}


export interface ProjectTable{
    checked: boolean;
    pid: number;
    name: string;
    description: string;
    createdBy: string;
    createdDate: Date;
    accessKey: String;
}
